### ISSUE TITLE
'array' validation rule passes when the value is an empty string and required_array_keys passes as long as at least 1 key exists

### ISSUE THREAD

**Issue by pintend** [laravel/framework#50292]
### Laravel Version

10.43.0 (Tinkerwell)

### PHP Version

8.2.16

### Description

TL;DR validation passes the validation rule of `array` when the data is an empty string

this happens because the `array` rule is not implicitly required so when the value is an empty string its not "validatable" becuase `presentOrRuleIsImplicit()` returns false on blank strings
https://github.com/laravel/framework/blob/4f7802dfc9993cb57cf69615491ce1a7eb2e9529/src/Illuminate/Validation/Validator.php#L645

check steps to reproduce for examples

---

I have an array which is required to be an array. if there are any items in the array, i need those to be an array as well

```php
// This is the data we need, inner items are optional, but each inner item can have its own structure
'outer' => [
    'inner1' => ['x' => 0, 'y' => 0, 'w' => 0, 'h' => 0]
    'inner2' => ['whatever']
]
```

lets set some rules
```php
[
    'outer' => 'array',
    'outer.*' => 'array',
    'outer.inner1' => 'nullable|array|required_array_keys:x,y,w,h',
    'outer.inner1.*' => 'required|numeric',
    'outer.inner2' => 'nullable|array|required_array_keys:something_else',
    'outer.inner2.*' => 'string'
]
```
using the above rules i would expect this data to fail
```php
'outer' => [
    'inner1' => '',
    ''
]
```
I should have
```php
'errors' => [
    'The outer.inner1 must be an array',
    'The outer.  0 must be an array.'
]
```
but instead it passes and i get a "not valid" array

so now if i do something like this
```php
isset($validated["outer"]["inner1"])
    ? $validated["outer"]["inner1"]["x"]
    : null;
```
i get a `TYPE ERROR  Cannot access offset of type string on string`

shouldn't `inner1` be validated as an array if it exists and not pass if its a string? 

I think it should be casted to null or removed from the validated array entirely

**Secondaly**
given the follwing input
```php
'outer' => [
    'inner1' => [],
]
```
you get the expected `The outer.inner1 field must contain entries for: x, y, w, h` validation error. 
but if you include any of the required keys `required_array_keys:x,y,w,h` then I don't get the validation error even though keys `y,w,h` are missing
```php
'outer' => [
    'inner1' => ['x'=>0],
]
```

### Steps To Reproduce

Try these examples in Tinkerwell, validation errors are expected but they pass.

```php
$validated = validator(
    [
        'outer' => [
            'inner1' => '',
            ''
        ]
    ],
    [
        'outer' => 'required|array',
        'outer.*' => 'array',
        'outer.inner1' => 'nullable|array|required_array_keys:x,y,w,h',
        'outer.inner1.*' => 'required|numeric',
        'outer.inner2' => 'nullable|array|required_array_keys:something_else',
        'outer.inner2.*' => 'string'
    ]
)->validated();

isset($validated['outer']['inner1'])
    ? $validated['outer']['inner1']['y']
    : null;
// TYPE ERROR  Cannot access offset of type string on string 
```

```php
$validated = validator(
    [
        'outer' => [
            'inner1' => ['x' => 0],
            ''
        ]
    ],
    [
        'outer' => 'required|array',
        'outer.*' => 'array',
        'outer.inner1' => 'nullable|array|required_array_keys:x,y,w,h',
        'outer.inner1.*' => 'required|numeric',
        'outer.inner2' => 'nullable|array|required_array_keys:something_else',
        'outer.inner2.*' => 'string'
    ]
)->validated();
// this should not pass as "y" is missing from the data and outer.0 is not an array, but it passes and returns
/*
[
    "outer" => [
      0 => "",
      "inner1" => [
        "x" => 0,
      ],
    ],
]
*/


isset($validated['outer']['inner1'])
    ? $validated['outer']['inner1']['y']
    : null;

// WARNING  Undefined array key "y"
```

```php
$validated = validator(
    [
        'outer' => [
            'inner1' => null,
        ]
    ],
    [
        'outer' => 'array',
        'outer.*' => 'array',
        'outer.inner1' => 'array',
    ]
)->validated();
// The outer.inner1 must be an array.


$validated = validator(
    [
        'outer' => [
            'inner1' => '',
        ]
    ],
    [
        'outer' => 'array',
        'outer.*' => 'array',
        'outer.inner1' => 'array',
    ]
)->validated();
// passes... why?
```

**Comment by crynobone** [comment]
Hey there,

While this may be a legitimate issue, can you first try posting your problem or question on one of the support channels below? If this issue can be definitively identified as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

* [Laracasts Forums](https://laracasts.com/discuss)
* [Laravel.io Forums](https://laravel.io/forum)
* [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
* [Discord](https://discordapp.com/invite/KxwQuKb)
* [Larachat](https://larachat.co/)
* [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!
